src/fontMetrics.js

```diff
@@ -226,21 +227,32 @@ const extraCharacterMap = {
     'я': 'r',
 };
 
+export type CharacterMetrics = {
+    depth: number;
+    height: number;
+    italic: number;
+    skew: number;
+    width: number;
+};
+
 /**
  * This function is a convenience function for looking up information in the
- * metricMap table. It takes a character as a string, and a style.
+ * metricMap table. It takes a character as a string, and a font.
  *
  * Note: the `width` property may be undefined if fontMetricsData.js wasn't
  * built using `Make extended_metrics`.
  */
-const getCharacterMetrics = function(character, style) {
+const getCharacterMetrics = function(
+    character: string,
+    font: string,
+): ?CharacterMetrics {
     let ch = character.charCodeAt(0);
     if (character[0] in extraCharacterMap) {
         ch = extraCharacterMap[character[0]].charCodeAt(0);
     } else if (cjkRegex.test(character[0])) {
         ch = 'M'.charCodeAt(0);
     }
-    const metrics = metricMap[style][ch];
+    const metrics = metricMap[font][ch];
     if (metrics) {
         return {
```

Comment:
Ah, sorry, I'd started porting this without paying attention to the existence of this commit. In any case, if you use a tuple type (that I commented on further below), flow will point out that `width: metrics[4]` is doing an out-of-bounds access.
So remove `width:` here. I grepped through the code base and I couldn't find any uses of it either. ;)

The now-closed [PR 853](https://github.com/Khan/KaTeX/pull/853) contains the work I did on `frontMetrics*` and `unicodeRegex`.